src/backend.ts

```diff
@@ -311,14 +321,69 @@ export function setupSuspendResumeHandler(): () => void {
 export function setupFocusChangeHandler(): () => void {
   let appIsStartingUp: boolean = false;
   let lastPid = 0;
+  let lastAppid = 0;
+  let validKeyEvent: SystemKeyEvent | null = null;
+  let keyEventFunction = (e: SystemKeyEvent) => {
+    const cancelDebouncedEvent = debounce(() => {
+      validKeyEvent = null;
+    }, 1000);
+    if (e.eKey === 0) {
```

Comment:
Ah, additionally `validKeyEvent` could be set to `null` in the `else` to be extra safe.

Reply:
```ts
    if (e.eKey === 0) {
      // Have not found any race condition issues with this approach since the key event fires long before focus change
      validKeyEvent = e;
      cancelDebouncedEvent();
    } else {
      cancelDebouncedEvent.cancel();
      validKeyEvent = null;
    }
```